lib/rabbit_feed/consumer_connection.rb

```diff
@@ -19,37 +18,35 @@ class ConsumerConnection
         },
     }.freeze
 
-    attr_reader :queue
-
-    def initialize channel
-      channel.prefetch(1) # Fetch one message at a time to preserve order
-      RabbitFeed.log.debug "Declaring queue on #{self.to_s} (channel #{channel.id}) named: #{RabbitFeed.configuration.queue} with options: #{queue_options}..."
+    def initialize
+      super
+      channel.prefetch(1)
       @queue = channel.queue RabbitFeed.configuration.queue, queue_options
       bind_on_accepted_routes
     end
 
-    def self.consume &block
-      with_connection do |consumer_connection|
-        consumer_connection.consume(&block)
-      end
-    end
-
     def consume &block
-      RabbitFeed.log.info "Consuming messages on #{self.to_s} from queue: #{RabbitFeed.configuration.queue}..."
-
-      consumer = queue.subscribe(SUBSCRIPTION_OPTIONS) do |delivery_info, properties, payload|
-        handle_message delivery_info, payload, &block
+      thread_safe do
```

Comment:
Also, and this is just a thought... if what you want to achieve is a single thread per the object of this class I would use abstractions that already do that... Like the Actor Model.... If you use `celluloid` and make the Consumer an Actor I think you could make your consumer's consume method guaranteed to be executed by one thread only always.. which seems like what you want